### ISSUE TITLE
Sentry integration

### ISSUE THREAD

**Issue by m-kuhn** [mapproxy/mapproxy#537]
For monitoring of a system, it would be nice to be able to send warnings, errors, exceptions to a [sentry](https://sentry.io) instance.

This could be implemented as a feature into the main code base and would introduce a python dependency [sentry_sdk](https://github.com/getsentry/sentry-python).
Would a pull request be accepted or should this be done in a deployment directly?

**Comment by arvkonstantin** [comment]
It's a good idea

**Comment by weskamm** [comment]
in my opinion this should not be included in mapproxy. The functionality needed here is not really a part of mapproxy. Why dont you use a middleware that keeps track of mapproxys messages and handles the forwarding of messages to your sentry instance?

**Comment by m-kuhn** [comment]
Would that be able to forward an uncaught python exception happening inside the mapproxy service?

**Comment by weskamm** [comment]
If the exception finds it way to the logfiles, why not?

**Comment by walkermatt** [comment]
> Would that be able to forward an uncaught python exception happening inside the mapproxy service?

Yes, I think that should be possible via WSGI middleware. In [this example](https://gist.github.com/walkermatt/1cae669defce037511ada4e0ed01b3cd#file-mapproxy_decorate-py-L59) the call to the MapProxy WSGI app instance could be wrapped in a `try`/ `except`.

**Comment by m-kuhn** [comment]
> > Would that be able to forward an uncaught python exception happening inside the mapproxy service?
> 
> Yes, I think that should be possible via WSGI middleware. In [this example](https://gist.github.com/walkermatt/1cae669defce037511ada4e0ed01b3cd#file-mapproxy_decorate-py-L59) the call to the MapProxy WSGI app instance could be wrapped in a `try`/ `except`.

Yep, correct. That was what I meant with "should this be done in a deployment directly" in the original post :+1: 
If that's the preferred way, this issue can be closed.
